React/Views/RCTScrollView.m

```diff
@@ -317,8 +317,8 @@ - (void)setFrame:(CGRect)frame
   CGSize boundsSize = self.bounds.size;
 
   self.contentOffset = CGPointMake(
-    MAX(0, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
-    MAX(0, MIN(originalOffset.y, fullContentSize.height - boundsSize.height)));
+    MAX(-contentInset.left, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
```

Comment:
I feel like contentOffset is kind of weird right now and should be deprecated or at least make it's behaviour clearer. The only use case it has is to set the initial scroll position because using refs with scrollTo will usually result in a flicker for a frame.

I was thinking changing it so something like `initialContentOffset` so it's clear it only affects the scroll position of the ScrollView when it's mounted and that changing the property after that won't have any effect. With these constraints do you think it would be possible to fix both issues? We'd probably have to find a way to wait for the ScrollView's content to be measured before adjusting contentOffset to the initial offset. I haven't worked with that code much recently so hopefully that makes sense.